prov/gni/src/gnix_dom.c

```diff
@@ -95,6 +95,8 @@ static int gnix_domain_close(fid_t fid)
 		/* TODO: free nic here */
 	}
 
+	atomic_dec(&domain->fabric->ref_cnt);
```

Comment:
You know what I'm going to say here, right?  Maybe we should add a macro or wrapper function to do the decrement and check that the original value was > 0?